File: src/main/java/com/devon/dapuzzo/show/Show.java
package com.devon.dapuzzo.show;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.NonNull;

import javax.persistence.*;
import java.io.Serializable;
import java.sql.Date;

/**
 * Created by devondapuzzo on 9/22/17.
 */
@Entity
@Data
@Table(name="all_shows")
@AllArgsConstructor
@NoArgsConstructor
public class Show implements Serializable {
    @Id
    @GeneratedValue
    private Integer id;
    @NonNull
    @Column(nullable = false)
    private Date date;
    @NonNull
    @Column(nullable = false)
    private String venue;
    @NonNull
    @Column(nullable = false)
    private String city;
    @NonNull
    @Column(nullable = false)
    private String state;
    @NonNull
    @Column(nullable = false)
    private String google_maps_link;
}

File: src/main/java/com/devon/dapuzzo/show/Show.kt
package com.devon.dapuzzo.show

import java.io.Serializable
import java.sql.Date
import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.Id
import javax.persistence.Table

@Entity
@Table(name = "all_shows")
data class Show(
        @Id
        val id: Int,
        @Column(nullable = false)
        val date: Date,
        @Column(nullable = false)
        val venue: String,
        @Column(nullable = false)
        val city: String,
        @Column(nullable = false)
        val state: String,
        @Column(nullable = false)
        val google_maps_link: String,
        @Column(nullable = false)
        val style: String
) : Serializable {
    constructor() : this(
            0,
            Date.valueOf("1999-01-10"),
            "DEFAULT",
            "DEFAULT",
            "DEFAULT",
            "DEFAULT",
            "DEFAULT"
    )
}